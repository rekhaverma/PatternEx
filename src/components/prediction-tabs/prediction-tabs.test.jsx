import React from 'react';
import PredictionTabs from './prediction-tabs.component';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<PredictionTabs />', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      className: 'test-class',
      activeTab: '1',
      tabs: [
        {
          count: 2,
          id: '1',
          title: 'tab 1 title',
        }, {
          count: 2,
          id: '2',
          title: 'tab 2 title',
        }],
      onClick: jest.fn(),
    };
    wrapper = shallow(<PredictionTabs {...props} />);
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClick when element is clicked', () => {
    wrapper.find('.test-class__el--active').simulate('click');
    expect(props.onClick.mock.calls.length).toBe(1);
  });
});
