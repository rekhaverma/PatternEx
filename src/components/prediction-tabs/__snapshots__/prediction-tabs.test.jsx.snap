// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PredictionTabs /> should match with snapshot 1`] = `
<div
  className="test-class"
>
  <div
    className="test-class__el--active"
    key="1"
    onClick={[Function]}
  >
    <span>
      2
    </span>
    <p>
      tab 1 title
    </p>
  </div>
  <div
    className="test-class__el"
    key="2"
    onClick={[Function]}
  >
    <span>
      2
    </span>
    <p>
      tab 2 title
    </p>
  </div>
</div>
`;
